packages/menu/src/presenters/MenuPresenter.js

```diff
@@ -108,6 +108,17 @@ const MenuPresenter = (props) => {
       });
 
       if (JSON.stringify(currentIds) !== JSON.stringify(prevIds)) {
+        const prevHighlightIndex = getHighlightIndex();
+
+        if (prevHighlightIndex !== 0) {
+          const selectedOptionId = prevIds[prevHighlightIndex - 1];
+          const currHighlightIndex = currentIds.findIndex(id => id === selectedOptionId);
+
+          if (currHighlightIndex !== -1) {
+            props.setHighlightIndex(currHighlightIndex + 1);
+          }
```

Comment:
@sunnut  In case the previous highlighted item was removed (`currHighlightIndex === -1`), should we reset the highlight index (set to 0) to avoid possible issues?